pov_tools/create_POV_data_files

```diff
@@ -0,0 +1,218 @@
+#!/bin/bash
+
+# create_POV_data_files
+
+cd "$(dirname "$0")"
+clear
+echo ""; echo "Welcome! Let's make some POV images!"; echo ""
+# replace whitespace in source filename with underscores
+find . -type f -name "* *" | while read file; do mv "$file" ${file// /_}; done
+have_logo=0
+height_only=0
+preset=0
+# get quantity of .png files and build array to print if multiples
+files=$(ls 2>/dev/null -Ubad1 -- *.png | wc -l)
+pngs=$(find . -type f -name '*.png')
+
+if [ "$files" -ge 2 ]; then
+	echo "OOPS! Too many .png image files found!"
+	echo ""
+	for strip in ${pngs[@]}; do
+		no_path="${strip##*/}"
+		echo ${no_path[*]}
+	done
+	echo ""
+	echo "There needs to be only ONE source .png image in the pov_tools folder."
+	echo ""
+	echo "Type 'r' - to remove ALL FILES with 'preview' in the name. "
+	echo "    If the additional filenames do not contain 'preview',"
+	echo "    manually delete all files except your source image."
+	echo ""
+	echo "Type q to quit."
+	read -p $'Press ENTER to start over.\n' fix
+	case $fix in
+		r) rm *preview*.png;
+			echo ""
+			echo "Files containing *preview* deleted if they existed"
+			read -n 1 -r -s -p $'Press any key to continue.\n'
+			exec "./create_POV_data_files";;
+		q) exit;;
+		*) exec "./create_POV_data_files";;
+	esac
+fi
+
+if [ "$files" -lt 1 ]; then
+	echo "OOPS! No .png image files found!"
+	echo ""
+	echo "Let's download an example Star Wars Logo for you to try :)"
+	echo "DOWNLOADING: 1024px-Star_Wars_Logo.svg.png .........."
+	echo ""
+	wget https://upload.wikimedia.org/wikipedia/commons/thumb/6/6c/Star_Wars_Logo.svg/1024px-Star_Wars_Logo.svg.png&>/dev/null &
```

Comment:
Why do you have an & at the end of this line?
It will make the wget command run in the background, so the file might not exist by the time we get to the next command.

Reply:
> ... when someone shows me a better way to do something, or offers a better understanding of how I should have done, I'm always willing to do the work to make the code better. It benefits me, it benefits the code, and ultimately, it's really not that much work.

I'm soaking it all in and really appreciate the professional guidance. I'll be back with fxes.

Reply:
> Why do you have an & at the end of this line? It will make the wget command run in the background, so the file might not exist by the time we get to the next command.

In the background intentional just to keep the console cleaner. The "fake downloading" line is there instead, and the pause until a key is pressed was enough time for it to download in testing. I suppose someone might have McDonalds WiFi and need more time.... I'll make it active.

Reply:
foreground